android/app/src/main/java/org/electroncash/electroncash3/Main.kt

```diff
@@ -654,15 +661,43 @@ class SeedPasswordDialog : PasswordDialog<SeedResult>() {
     }
 }
 
-
 class SeedDialog : AlertDialogFragment() {
     override fun onBuildDialog(builder: AlertDialog.Builder) {
         builder.setTitle(R.string.Wallet_seed)
-            .setView(R.layout.wallet_new_2)
-            .setPositiveButton(android.R.string.ok, null)
+                .setView(R.layout.wallet_new_2)
+                .setPositiveButton(android.R.string.ok, null)
     }
 
     override fun onShowDialog() {
         setupSeedDialog(this)
     }
 }
+
+class WalletInformationDialog : AlertDialogFragment() {
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+
+        builder.setView(R.layout.wallet_information)
+            .setPositiveButton(android.R.string.ok, null)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        fabCopyMasterKey2.setOnClickListener {
+            val textToCopy = walletMasterKey.text
+            copyToClipboard(textToCopy, R.string.Master_public)
+        }
+    }
+
+    override fun onShowDialog() {
+        super.onShowDialog()
+
+        val masterKey = daemonModel.commands.callAttr("getmpk").toString()
```

Comment:
This crashes with an imported address or private key wallet. I think in that kind of wallet, the public key box and the copy button should be hidden.